pilot/cmd/pilot-agent/main.go

```diff
@@ -135,9 +135,18 @@ var (
 			}
 
 			// Obtain all the IPs from the node
-			if ipAddr, ok := proxy.GetPrivateIPs(context.Background()); ok {
-				log.Infof("Obtained private IP %v", ipAddr)
-				role.IPAddresses = append(role.IPAddresses, ipAddr...)
+			if ipAddrs, ok := proxy.GetPrivateIPs(context.Background()); ok {
+				log.Infof("Obtained private IP %v", ipAddrs)
+				if len(role.IPAddresses) == 1 {
```

Comment:
This is not sufficient for de-duping IP addresses.  We should assume that both `role.IPAddresses` and `ipAddrs` could each have duplicates.  You should probably de-dup using a set.  I think it might be best to leave this code as-is and perform the de-duping in the `bootstrap_config.go`.

Reply:
I donot think this is needed, before these compare the `role.IPAddresses` has no more than 1 ips.  You can check its initialization.

Reply:
Agreed, the logic will restrict the length of `role.IPAddresses` to either 0 or 1 currently.

However, making that assumption in this code is fragile ... future changes to the logic above could easily unknowingly re-introduce duplicates.

The logic in `bootstrap_config.go` requires that there are no duplicates ... so it should just do the de-duplication logic there.

Reply:
ok